cibuildwheel/platforms/__init__.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Final, Protocol, assert_never
+
+from cibuildwheel.architecture import Architecture
+from cibuildwheel.options import Options
+from cibuildwheel.platforms import linux, macos, pyodide, windows
+from cibuildwheel.selector import BuildSelector
+from cibuildwheel.typing import GenericPythonConfiguration, PlatformName
+
+
+class PlatformModule(Protocol):
+    # note that as per PEP544, the self argument is ignored when the protocol
+    # is applied to a module
+    def all_python_configurations(self) -> Sequence[GenericPythonConfiguration]: ...
+
+    def get_python_configurations(
+        self, build_selector: BuildSelector, architectures: set[Architecture]
+    ) -> Sequence[GenericPythonConfiguration]: ...
+
+    def build(self, options: Options, tmp_path: Path) -> None: ...
+
+
+def get_platform_module(platform: PlatformName) -> PlatformModule:
+    if platform == "linux":
+        return linux
+    if platform == "windows":
+        return windows
+    if platform == "macos":
+        return macos
+    if platform == "pyodide":
+        return pyodide
+    assert_never(platform)
```

Comment:
This function is probably not needed anymore given it's just a look-up in the ALL_PLATFORM_MODULES dict now ?

Reply:
Done!